packages/@react-spectrum/table/src/TableView.tsx

```diff
@@ -288,13 +288,19 @@ function TableView<T extends object>(props: SpectrumTableProps<T>, ref: DOMRef<H
     }
   };
 
-  let isVerticalScrollbarVisible = false;
-  let isHorizontalScrollbarVisible = false;
-  if (bodyRef.current && domRef.current) {
-    // 2 is the width of the border which is not part of the box size
-    isVerticalScrollbarVisible = domRef.current.getBoundingClientRect().width > bodyRef.current.children[0]?.getBoundingClientRect().width + 2;
-    isHorizontalScrollbarVisible = domRef.current.getBoundingClientRect().height > bodyRef.current.children[0]?.getBoundingClientRect().height + 2;
-  }
+  let [isVerticalScrollbarVisible, setVerticalScollbarVisible] = useState(false);
+  let [isHorizontalScrollbarVisible, setHorizontalScollbarVisible] = useState(false);
+  let viewport = useRef({x: 0, y: 0, width: 0, height: 0});
+  let onVisibleRectChange = useCallback((e) => {
+    if (viewport.current.width === e.width && viewport.current.height === e.height) {
+      return;
+    }
+    viewport.current = e;
+    if (bodyRef.current) {
+      setVerticalScollbarVisible(bodyRef.current.clientWidth + 2 < bodyRef.current.offsetWidth);
+      setHorizontalScollbarVisible(bodyRef.current.clientHeight + 2 < bodyRef.current.offsetHeight);
+    }
+  }, []);
```

Comment:
curious, mind explaining what necessitated the need to do this onVisibleRectChange stuff but ListView only needed a layout effect?

Reply:
One component renders more frequently, i'm not sure why
I could hook it up to on visible rect change for both, but it would mean adding that as a prop to Virtualizer, which I didn't want to do

Reply:
bleh, thats unfortunate. If this is the most reliable way to get the correct TableView width and height then this is fine.